### PR TITLE
LOCATE  func

### LINKED ISSUE #3060: LOCATE() | Return the position of the first occurrence of substring

### PR DESCRIPTION
**Pull request by mshauneu**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Impl of LOCATE() func

## Changelog

- New Feature

## Related Issues

Fixes #3060 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/strings/locate.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                }
+            }
+            (
+                DataColumn::Array(ss_series),
+                DataColumn::Array(s_series),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let Some(p) = p {
+                    DFUInt64Array::from_iter(
+                        ss_series
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/locate.rs:126]: `into_no_null_iter` ignore the null values.

We can use `DFUInt64Array::new_from_iter_validity` to extends the validity of `ss_series`.

Comment (mshauneu) [reply common/functions/src/scalars/strings/locate.rs]: Fixed...

**Comment by BohuTANG** [comment]
@mergify update

File: common/functions/src/scalars/strings/locate.rs [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            (
+                DataColumn::Constant(DataValue::String(ss), _),
+                DataColumn::Array(s_series),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let (Some(ss), Some(p)) = (ss, p) {
+                    let s_array = s_series.string()?;
+                    DFUInt64Array::new_from_iter_validity(
+                        s_array.into_no_null_iter().map(|s| find_at(s, &ss, &p)),
+                        combine_validities(s_array.inner().validity(), None),
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/locate.rs:94]: Why not just use `s_array.inner().validity().clone()` ?

**Comment by BohuTANG** [comment]
@mergifyio update

File: common/functions/src/scalars/strings/locate.rs (hunk 2 above)
Comment (mshauneu) [reply common/functions/src/scalars/strings/locate.rs]: `combine_validities` do it internally, anyway changed to `inner().validity().cloned()`...

**Review by sundy-li — APPROVED** [review]

File: website/databend/mkdocs.yml [hunk 3]
```diff
@@ -225,6 +225,7 @@ nav:
               - SUBSTRING: sqlstatement/string-functions/substring.md
               - TO_BASE64: sqlstatement/string-functions/to_base64.md
               - FROM_BASE64: sqlstatement/string-functions/from_base64.md
+              - LOCATE: sqlstatement/string-functions/locate.md
```
Comment (BohuTANG) [inline website/databend/mkdocs.yml:228]: 💯

**Pull request merged**